test/python/circuit/test_circuit_load_from_qasm.py

```diff
@@ -48,24 +48,117 @@ def test_qasm_file(self):
         q_circuit_2.measure(qr_b, cr_d)
         self.assertEqual(q_circuit, q_circuit_2)
 
+    def test_loading_all_qelib1_gates(self):
+        """Test setting up a circuit with all gates defined in qiskit/qasm/libs/qelib1.inc."""
+        qasm_str = '\n'.join(["OPENQASM 2.0;",
```

Comment:
Could you have just loaded this from all_gates.qasm which you added?

Reply:
I would prefer to keep it as string in the same function, so we don't have to modify code in several places. The `all_gates.qasm` should've been removed, but slipped through my fingers. 

Could `all_gates.qasm` be useful in other places? Then it would of course change things and keeping it seems good.

Reply:
I think it's better to have it as string in the same function, so all is kept in one place. The `all_gates.qasm` should've been removed but managed to slip through 😉 

If `all_gates.qasm` could be handy for other tests however, that changes things. Then keeping it seems better.

Reply:
The issue I see is that this seems to add to the number of places where something has to change if qelib1 is updated. To reduce that I think it would be good to either remove it here or remove all_gates.qasm since nothing else seems to use that currently.

Reply:
Done in e5ef275.